openid-4-verifiable-presentations-1_0.md

```diff
@@ -1817,13 +1829,20 @@ precautions against leaking information about the claim value when processing
 In both cases listed here, it needs to be considered that returning an error
 response can also leak information about the processing outcome of `values`.
 
-# Privacy Considerations
+## User Consent and Choice {#user_consent_and_choice}
 
-Many privacy considerations are specific to the credential format and associated proof type used in any particular presentation.
-This section focuses on privacy considerations that are specific to the presentation protocol with some treatment also
-given to considerations that apply across some common credential formats.
+* Wallets should obtain explicit, informed consent from the End-User before releasing any Verifiable Credential or Presentation to a Verifier.
+* The Wallet should display the purpose and requested claims clearly, especially when `transaction_data` or `claim_sets` are used to express sensitive queries.
+* If an error prevents the Wallet from honoring a request (e.g., missing credentials or mismatched claim values), the Wallet should inform the user in a privacy-preserving way.
+
+## Purpose Legitimacy and Specification {#purpose_legitimacy_and_specification}
 
-## Selective Disclosure
+* The verifier ensures that the purposes are sufficiently specific and are communicated before the time the information is collected. For example, it is shown before the presentation request is sent or is sent in the request to the wallet. In the latter case, the wallet communicates them with the user before the user can make their choice. 
+* The verifier should ensure that the purpose complies with applicable law and relies on a permissible legal basis. 
```

Comment:
Is reminding verifiers not to break the law needed? I think delving into legal recommendations is not in the scope of this spec.

Reply:
This is not a legal recommendation.

Reply:
```suggestion
```
we talk about trust frameworks and policies but we have explicitly avoided talking about laws